collector/runtime-benchmarks/bufreader/src/main.rs

```diff
@@ -0,0 +1,30 @@
+use benchlib::benchmark::BenchmarkSuite;
+use snap::{read::FrameDecoder, write::FrameEncoder};
+use std::io::{BufRead, BufReader, Write};
+
+const BYTES: usize = 64 * 1024 * 1024;
+
+fn main() {
+    let mut suite = BenchmarkSuite::new();
+
+    suite.register("bufreader-snappy", || {
+        let data = vec![0u8; BYTES];
+        move || {
+            let mut compressed = Vec::new();
+            FrameEncoder::new(&mut compressed)
+                .write_all(&data[..])
+                .unwrap();
+            let mut reader = BufReader::with_capacity(BYTES, FrameDecoder::new(&compressed[..]));
+
+            while let Ok(buf) = reader.fill_buf() {
+                if buf.is_empty() {
+                    break;
+                }
+                let len = buf.len();
+                reader.consume(len);
```

Comment:
I don't suppose that it wouldn't be optimized out, but maybe we could return something computed from the closure? Anything returned will be turned into a `black_box`. But I'm not sure if it's that important here.

Reply:
I added a `black_box` call in the loop, because in real code, we'd probably do some complicated operation with the buffer inside the loop. Does that sound good to you?

Reply:
Yes, thank you!